--- a/solenv/bin/concat-deps.c
+++ b/solenv/bin/concat-deps.c
@@ -1091,7 +1091,7 @@
                     created_line = generate_phony_line(src_relative, "o");
                     rc = generate_phony_file(fn, created_line);
                 }
-                else if(strncmp(src_relative, "GenAsmObject/", 14) == 0)
+                else if(strncmp(src_relative, "GenAsmObject/", 13) == 0)
                 {
                     created_line = generate_phony_line(src_relative, "o");
                     rc = generate_phony_file(fn, created_line);
--- a/solenv/gbuild/platform/EMSCRIPTEN_INTEL_GCC.mk
+++ b/solenv/gbuild/platform/EMSCRIPTEN_INTEL_GCC.mk
@@ -8,6 +8,10 @@
 #
 
 include $(GBUILDDIR)/platform/unxgcc.mk
+
+ifeq ($(OS_FOR_BUILD),MACOSX)
+gb_Library_PLAINEXT_FOR_BUILD := .dylib
+endif
 
 gb_RUN_CONFIGURE := $(SRCDIR)/solenv/bin/run-configure
 # avoid -s SAFE_HEAP=1 - c.f. gh#8584 this breaks source maps
--- a/solenv/gbuild/LinkTarget.mk
+++ b/solenv/gbuild/LinkTarget.mk
@@ -646,7 +646,7 @@
 gb_GenAsmObject_get_source = $(call gb_AsmObject_get_source,$(WORKDIR),$(1))
 
 $(call gb_GenAsmObject_get_target,%) :
-	$(call gb_AsmObject__command,$@,$*,$<,$(call gb_GenAsmObject_get_dep_target,$*))
+	$(call gb_AsmObject__command,$@,$*,$(call gb_GenAsmObject_get_source,$*),$(call gb_GenAsmObject_get_dep_target,$*))
 
 ifeq ($(gb_FULLDEPS),$(true))
 $(dir $(call gb_GenAsmObject_get_dep_target,%)).dir :
--- a/solenv/gbuild/Conditions.mk
+++ b/solenv/gbuild/Conditions.mk
@@ -13,7 +13,7 @@
 # or the $(2) multiple times.
 
 define gb_CondExeLockfile
-$(if $(and $(filter-out ANDROID MACOSX iOS WNT,$(OS))),$(1),$(2))
+$(if $(or $(filter-out ANDROID MACOSX iOS WNT,$(OS)),$(and $(filter MACOSX,$(OS)),$(filter EMSCRIPTEN,$(BUILD_TYPE_FOR_HOST)))),$(1),$(2))
 endef
 
 define gb_CondExeRegistryTools
