--- a/external/openssl/ExternalProject_openssl.mk
+++ b/external/openssl/ExternalProject_openssl.mk
@@ -52,7 +52,7 @@
             $(if $(filter X86_64,$(CPUNAME)),darwin64-x86_64-cc)\
             $(if $(filter AARCH64,$(CPUNAME)),darwin64-arm64-cc)\
 	  ,\
-	    $(if $(filter EMSCRIPTEN,$(OS)),no-engine no-dso no-dgram no-srtp no-err no-ocsp no-psk no-ts no-asm) \
+	    $(if $(filter EMSCRIPTEN,$(OS)),gcc -pthread no-engine no-dso no-dgram no-srtp no-err no-ocsp no-psk no-ts no-asm) \
           )\
         )\
       )\
@@ -79,7 +79,7 @@
 	$(call gb_Trace_StartRange,openssl,EXTERNAL)
 	$(call gb_ExternalProject_run,build,\
 		unset MAKEFLAGS && \
-		$(if $(filter LINUX MACOSX FREEBSD ANDROID SOLARIS iOS,$(OS)), \
+		$(if $(filter LINUX MACOSX FREEBSD ANDROID SOLARIS iOS EMSCRIPTEN,$(OS)), \
 			./Configure, \
 		$(if $(filter WNT,$(OS)), \
 			$(PERL) Configure, \
--- a/external/argon2/ExternalProject_argon2.mk
+++ b/external/argon2/ExternalProject_argon2.mk
@@ -43,7 +43,7 @@
 	$(call gb_Trace_StartRange,argon2,EXTERNAL)
 	$(call gb_ExternalProject_run,build,\
 		CFLAGS="$(CFLAGS) $(if $(filter-out WNT MACOSX,$(OS)),-fvisibility=hidden) -fPIC" \
-		MAKEFLAGS= $(MAKE) \
+		MAKEFLAGS= $(MAKE) $(if $(filter EMSCRIPTEN,$(OS)),libargon2.a) \
 			OPTTARGET=$(if $(filter X86_64,$(CPUNAME)),x86-64,forcefail) \
 		$(if $(CROSS_COMPILING),,&& $(MAKE) test) \
 	)
--- a/external/argon2/UnpackedTarball_argon2.mk
+++ b/external/argon2/UnpackedTarball_argon2.mk
@@ -15,6 +15,7 @@
 	external/argon2/0001-Fix-possible-compiler-error-due-to-undefined-_MSC_VE.patch \
 	$(if $(filter WNT_AARCH64,$(OS)_$(CPUNAME)),external/argon2/0002-Add-WinARM64-vcxproj-config.patch) \
 	external/argon2/private-symbols.patch.0 \
+	external/argon2/use-configured-archiver.patch.0 \
 ))
 
 # vim: set noet sw=4 ts=4:
--- /dev/null
+++ b/external/argon2/use-configured-archiver.patch.0
@@ -0,0 +1,11 @@
+--- Makefile
++++ Makefile
+@@ -182,7 +182,7 @@
+ 		$(CC) $(CFLAGS) $(LIB_CFLAGS) $(LDFLAGS) $(SO_LDFLAGS) $^ -o $@
+ 
+ $(LIB_ST): 	$(OBJ)
+-		ar rcs $@ $^
++		$(AR) rcs $@ $^
+ 
+ .PHONY: clean
+ clean:
