diff --git a/external/libxml2/ExternalProject_libxml2.mk b/external/libxml2/ExternalProject_libxml2.mk
--- a/external/libxml2/ExternalProject_libxml2.mk
+++ b/external/libxml2/ExternalProject_libxml2.mk
@@ -31,6 +31,8 @@
 			$(if $(filter 17.%,$(VCVER)),-G "Visual Studio 17 2022") \
 			$(if $(filter 18.%,$(VCVER)),-G "Visual Studio 18 2026") \
 			-A $(gb_MSBUILD_PLATFORM) \
+			-DCMAKE_POLICY_DEFAULT_CMP0141=NEW \
+			-DCMAKE_MSVC_DEBUG_INFORMATION_FORMAT=Embedded \
 			-DLIBXML2_WITH_TESTS=OFF \
 			-DLIBXML2_WITH_ICONV=OFF \
 			-DLIBXML2_WITH_PYTHON=OFF \
