--- a/svl/Library_svl.mk
+++ b/svl/Library_svl.mk
@@ -29,7 +29,7 @@
     icuuc \
     mdds_headers \
     libxml2 \
-    zxcvbn-c \
+    $(if $(DISABLE_DYNLOADING),,zxcvbn-c) \
 ))
 
 $(eval $(call gb_Library_set_componentfile,svl,svl/util/svl,services))
--- a/svl/source/misc/PasswordHelper.cxx
+++ b/svl/source/misc/PasswordHelper.cxx
@@ -25,7 +25,9 @@
 #include <unicode/regex.h>
 #include <unicode/unistr.h>
 #include <unicode/errorcode.h>
+#if !defined(DISABLE_DYNLOADING)
 #include <zxcvbn.h>
+#endif
 #include <sal/log.hxx>
 
 #include <com/sun/star/uno/Sequence.hxx>
@@ -136,6 +138,7 @@
     return bResult;
 }
 
+#if !defined(DISABLE_DYNLOADING)
 double SvPasswordHelper::GetPasswordStrengthPercentage(const char* pPassword)
 {
     // Entropy bits ≥ 112 are mapped to 100% password strength.
@@ -152,6 +155,8 @@
     OString aPasswordUtf8 = aPassword.toUtf8();
     return GetPasswordStrengthPercentage(aPasswordUtf8.getStr());
 }
+
+#endif
 
 bool SvPasswordHelper::PasswordMeetsPolicy(const char* pPassword,
                                            const std::optional<OUString>& oPasswordPolicy)
--- a/sfx2/source/dialog/passwd.cxx
+++ b/sfx2/source/dialog/passwd.cxx
@@ -47,8 +47,12 @@
     // if there's a confirm entry, the dialog is being used for setting a password
     if (m_xConfirm1ED->get_visible())
     {
+#if defined(DISABLE_DYNLOADING)
+        m_xPassword1StrengthBar->set_visible(false);
+#else
         m_xPassword1StrengthBar->set_percentage(
             SvPasswordHelper::GetPasswordStrengthPercentage(aPassword1Text));
+#endif
         bool bPasswordMeetsPolicy = SvPasswordHelper::PasswordMeetsPolicy(
             aPassword1Text, moPasswordPolicy);
         m_xPassword1ED->set_message_type(bPasswordMeetsPolicy ? weld::EntryMessageType::Normal
@@ -61,8 +65,12 @@
     {
         OUString aPassword2Text = m_xPassword2ED->get_text();
 
+#if defined(DISABLE_DYNLOADING)
+        m_xPassword2StrengthBar->set_visible(false);
+#else
         m_xPassword2StrengthBar->set_percentage(
             SvPasswordHelper::GetPasswordStrengthPercentage(m_xPassword2ED->get_text()));
+#endif
 
         // second password is optional, ignore policy if it is empty
         bool bPasswordMeetsPolicy
--- a/cui/source/dialogs/passwdomdlg.cxx
+++ b/cui/source/dialogs/passwdomdlg.cxx
@@ -131,10 +131,15 @@
     }
 
     // if password doesn't meet policy cap the percentage at 70%
+#if defined(DISABLE_DYNLOADING)
+    if (pLevelBar)
+        pLevelBar->set_visible(false);
+#else
     if (pLevelBar)
         pLevelBar->set_percentage(
             std::min(SvPasswordHelper::GetPasswordStrengthPercentage(aPasswordText),
                      bPasswordMeetsPolicy ? std::numeric_limits<double>::max() : 70.0));
+#endif
 
     if (m_nMaxPasswdLen)
     {
--- a/uui/source/masterpasscrtdlg.cxx
+++ b/uui/source/masterpasscrtdlg.cxx
@@ -35,8 +35,12 @@
 {
     OUString aPasswordText = m_xEDMasterPasswordCrt->get_text();
     m_xOKBtn->set_sensitive(aPasswordText.getLength() >= 1);
+#if defined(DISABLE_DYNLOADING)
+    m_xPasswdStrengthBar->set_visible(false);
+#else
     m_xPasswdStrengthBar->set_percentage(
         SvPasswordHelper::GetPasswordStrengthPercentage(aPasswordText));
+#endif
 
     if(m_oPasswordPolicy)
     {
--- a/sc/source/ui/miscdlgs/protectiondlg.cxx
+++ b/sc/source/ui/miscdlgs/protectiondlg.cxx
@@ -162,8 +162,12 @@
     OUString aPass1 = m_xPassword1Edit->get_text();
     if (&rEntry == m_xPassword1Edit.get())
     {
-        m_xPasswordStrengthBar->set_percentage(
+#if defined(DISABLE_DYNLOADING)
+        m_xPasswordStrengthBar->set_visible(false);
+#else
+        m_xPasswordStrengthBar->set_percentage(
             SvPasswordHelper::GetPasswordStrengthPercentage(aPass1));
+#endif
     }
     OUString aPass2 = m_xPassword2Edit->get_text();
     m_xBtnOk->set_sensitive(aPass1 == aPass2);
